File: src/org/musicbrainz/barcodescanner/BaseActivity.java
/*
 * Copyright (C) 2012 Philipp Wolfer <ph.wolfer@googlemail.com>
 * 
 * This file is part of MusicBrainz Picard Barcode Scanner.
 * 
 * MusicBrainz Picard Barcode Scanner is free software: you can redistribute it
 * and/or modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the License,
 * or (at your option) any later version.
 * 
 * MusicBrainz Picard Barcode Scanner is distributed in the hope that it will be
 * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
 * Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License along with
 * MusicBrainz Picard Barcode Scanner. If not, see
 * <http://www.gnu.org/licenses/>.
 */

package org.musicbrainz.barcodescanner;

import com.markupartist.android.widget.ActionBar;

import android.app.Activity;
import android.content.Intent;
import android.view.ViewStub;

public abstract class BaseActivity extends Activity {

	protected void setSubView(int subView) {
		setContentView(R.layout.main);
		ViewStub content = (ViewStub) findViewById(R.id.view_content);
		content.setLayoutResource(subView);
		content.inflate();

		ActionBar actionBar = (ActionBar) findViewById(R.id.actionbar);
		actionBar.setHomeAction(new ActionBar.IntentAction(this, new Intent(
				this, ConnectActivity.class), R.drawable.ic_menu_tags));
		// actionBar.setHomeLogo(R.drawable.icon);
		// actionBar.setHomeAction(new IntentAction(this, HomeActivity
		// .createIntent(this), R.drawable.ic_launcher));
		// actionBar.addAction(new IntentAction(this, createShareIntent(),
		// R.drawable.ic_title_share_default));
		// actionBar.addAction(new ToastAction());
	}

}

File: PicardBarcodeScanner/src/main/java/org/musicbrainz/picard/barcodescanner/activities/BaseActivity.kt
/*
 * Copyright (C) 2012 Philipp Wolfer <ph.wolfer@googlemail.com>
 * 
 * This file is part of MusicBrainz Picard Barcode Scanner.
 * 
 * MusicBrainz Picard Barcode Scanner is free software: you can redistribute it
 * and/or modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the License,
 * or (at your option) any later version.
 * 
 * MusicBrainz Picard Barcode Scanner is distributed in the hope that it will be
 * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
 * Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License along with
 * MusicBrainz Picard Barcode Scanner. If not, see
 * <http://www.gnu.org/licenses/>.
 */
package org.musicbrainz.picard.barcodescanner.activities

import android.content.Intent
import android.content.pm.ApplicationInfo
import android.os.Build
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.ViewStub
import androidx.appcompat.app.AppCompatActivity
import org.musicbrainz.picard.barcodescanner.R
import org.musicbrainz.picard.barcodescanner.util.Preferences

abstract class BaseActivity : AppCompatActivity() {
    private var mPreferences: Preferences? = null
    protected fun setSubView(subView: Int) {
        setContentView(R.layout.main)
        val content = findViewById<View>(R.id.view_content) as ViewStub
        content.layoutResource = subView
        content.inflate()
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        handleIntents()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu items for use in the action bar
        val inflater = menuInflater
        inflater.inflate(R.menu.main_activity_actions, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle presses on the action bar items
        return when (item.itemId) {
            R.id.action_settings -> {
                val preferencesIntent = Intent(
                    this,
                    PreferencesActivity::class.java
                )
                startActivity(preferencesIntent)
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    protected open fun handleIntents() {}
    protected val preferences: Preferences
        get() {
            if (mPreferences == null) mPreferences = Preferences(this)
            return mPreferences!!
        }
    protected val isRunningInEmulator: Boolean
        get() = ApplicationInfo.FLAG_DEBUGGABLE != 0 &&
                ("google_sdk" == Build.PRODUCT || "sdk" == Build.PRODUCT || "sdk_x86" == Build.PRODUCT || "sdk_gphone" == Build.PRODUCT || "sdk_gphone_x86" == Build.PRODUCT)
}